commit d6e57967495107d6c85880f8514bed5c6d20e9d6
Author: Christian Hesse <mail@eworm.de>
Date:   Mon Dec 20 10:36:50 2021 +0100

    unescape when extracting link
    
    The file name is escaped when archiving, so it needs to be unescaped
    when extracting.

diff --git a/libarchive/archive_read_support_format_mtree.c b/libarchive/archive_read_support_format_mtree.c
index 9c46cfae..88bca76f 100644
--- a/libarchive/archive_read_support_format_mtree.c
+++ b/libarchive/archive_read_support_format_mtree.c
@@ -1675,6 +1675,7 @@ parse_keyword(struct archive_read *a, struct mtree *mtree,
 		break;
 	case 'l':
 		if (strcmp(key, "link") == 0) {
+			parse_escapes(val, NULL);
 			archive_entry_copy_symlink(entry, val);
 			return (ARCHIVE_OK);
 		}
